### PR TITLE
Add playbook to deploy stateless dogwood instances

### PR DESCRIPTION
**Pull request by omarkhan**
We currently use the `edx_sandbox.yml` playbook to deploy all our sandboxes and production instances. While this kind of works, the playbook was never meant to be used to install stateless production instances, and we've already had upstream contributions rejected because they weren't in line with the intended use of the playbook. Moreover, the playbook contains more services than we need on a typical production instance.

This pull request adds a new `edx_production.yml` playbook aimed specifically at our use case: Deploy a production dogwood instance that uses existing MySQL and MongoDB services.
#### Differences with the `edx_sandbox.yml` playbook
- Services are split into separate plays with different group names, so that service deployment can be defined in inventory.
- The `mysql`, `edxlocal` and `mongo` roles are excluded. It is expected that production deployments will use external databases that have been configured separately.
- The `analytics_api`, `insights`, `edx_notes_api`, `demo`, `edx_ansible` roles are excluded.

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @omarkhan! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

To automatically create an OSPR issue for this pull request, just visit this link: https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fconfiguration&number=3118

File: playbooks/edx-stateless.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    # Certificate generation
+    # https://github.com/edx/edx-certificates
+    - certs
+
+    # Ecommerce (optional)
+    # https://github.com/edx/ecommerce
+    - role: nginx
+      nginx_sites:
+      - ecommerce
+      when: SANDBOX_ENABLE_ECOMMERCE
```
Comment (bradenmacdonald) [inline playbooks/edx-stateless.yml]: I think this var should be renamed since this isn't a SANDBOX.

Comment (omarkhan) [inline playbooks/edx-stateless.yml]: Done: https://github.com/edx/configuration/pull/3118/commits/afd0098e71b1a62dc346392e130e38745dc275f4

**Comment by mtyaka** [comment]
@omarkhan Can we add the `postfix_queue` role?

File: playbooks/edx-stateless.yml [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    EDXAPP_LMS_NGINX_PORT: '80'
+    ENABLE_ECOMMERCE: False  # Disable ecommerce by default
+
+  roles:
+
+    # Server setup
+    - swapfile
+
+    # RabbitMQ (TODO: remove this)
+    - role: rabbitmq
```
Comment (feanil) [inline playbooks/edx-stateless.yml]: Why not remove it now?  This application holds data and so isn't really stateless.  It also has complexity with scaling horizontally so it doesn't really seem like it belongs here.

File: playbooks/edx-stateless.yml [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      rabbitmq_ip: 127.0.0.1
+
+    # Elasticsearch (TODO: remove this)
```
Comment (feanil) [inline playbooks/edx-stateless.yml]: Same question as above.

**Comment by feanil** [comment]
@omarkhan this change got bigger and seems to have changed approach, what's going on?

**Comment by omarkhan** [comment]
@feanil sorry this is very much a work in progress. We are trying to find the best way to write a playbook that meets our needs for a production setup.

**Comment by bradenmacdonald** [comment]
@feanil Do you have a PR for the playbook that you showed at the conference hackathon? I need to figure out if that supersedes this one, and how to proceed.

**Comment by feanil** [comment]
@bradenmacdonald The branch is here: https://github.com/edx/configuration/tree/feanil/small_deployments

It's not a PR because there are still some outstanding issues around connecting between the two machines that are unresolved.

**Comment by edx-webhook** [comment]
Thanks for the pull request, @omarkhan! I've created [OSPR-2097](https://openedx.atlassian.net/browse/OSPR-2097) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



If you like, you can add yourself to the [AUTHORS](https://github.com/edx/configuration/blob/master/AUTHORS)      file for this repo, though that isn't required.     Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

**Pull request closed without merging**